src/main/kotlin/no/nav/familie/ef/sak/vedtak/PatchPeriodetypeBarnetilsynController.kt

```diff
@@ -0,0 +1,55 @@
+package no.nav.familie.ef.sak.vedtak
+
+import no.nav.familie.ef.sak.vedtak.domain.BarnetilsynWrapper
+import no.nav.familie.ef.sak.vedtak.domain.Periodetype
+import no.nav.familie.ef.sak.vedtak.dto.ResultatType
+import no.nav.security.token.support.core.api.Unprotected
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.http.MediaType
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping(path = ["/api/periodetype-barnetilsyn-patch"], produces = [MediaType.APPLICATION_JSON_VALUE])
+@Unprotected
+@Validated
+class PatchPeriodetypeBarnetilsynController(
+    private val vedtakRepository: VedtakRepository
+) {
+
+    val logger: Logger = LoggerFactory.getLogger(this::class.java)
+
+    @Transactional
+    @PostMapping("{dryRun}")
+    fun patchPeriodetyperBarnetilsyn(@PathVariable dryRun: Boolean) {
+
+        val barnetilsynSaker = vedtakRepository.findAll().filter { it.barnetilsyn != null }
+
+        if (barnetilsynSaker.any { it.resultatType == ResultatType.SANKSJONERE }) {
+            error("Skal ikke finnes sanksjon av barnetilsyn")
+        }
+
+        val patchetSaker = barnetilsynSaker.map { vedtak ->
+            val nyePerioder = vedtak.barnetilsyn?.perioder?.map { periode ->
+                val nyPeriode = periode.copy(periodetype = if (periode.erMidlertidigOpphør == true) Periodetype.OPPHØR else Periodetype.ORDINÆR)
+                logger.info("Periode: $periode")
+                logger.info("Patchet: $nyPeriode")
```

Comment:
Vet ikke helt hva dette skal gi? Vi legger på periodetype? 👀

Reply:
Noen etterspurte logging av hvordan patchen vil se ut, så tenkte i hvert fall å logge når jeg kjører en dryrun.

Kanksje ikke helt hensiktmessig å spamme loggen med alle perioder i prod, men tenker dette er en ekstra sikkerthet for å verifisere at ting går bra når vi patcher.

Tenker du det er noen grunner til å ikke logge? :)

Reply:
Har vanskelig for å tro at den skal gi noe verdi når vi oppdaterer et objekt med ett fixed verdi :) 
Logger det samme objektet, med 1 nytt verdi

Reply:
Ja, egentlig enig med deg. Fjerner det. Noen forslag til en god måte å få verifisert en dry run?